File: android/src/main/kotlin/com/abom/applocal/app_lang/AppLangPlugin.kt
package com.abom.applocal.app_lang

import androidx.annotation.NonNull

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import android.os.Build
import java.util.Locale


/** AppLangPlugin */
class AppLangPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "app_lang")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    if (call.method == "getPhoneLanguage") {
      val language = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
        Locale.getDefault().toLanguageTag()
      } else {
        Locale.getDefault().language
      }
      result.success(language)
    } else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}

File: ios/Classes/AppLangPlugin.swift
import Flutter
import UIKit

public class AppLangPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "app_lang", binaryMessenger: registrar.messenger())
    let instance = AppLangPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPhoneLanguage":
      let language = Locale.preferredLanguages.first
      result(language)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
